similar/arch/ogl/ogl.cpp

```diff
@@ -1329,6 +1338,12 @@ void ogl_stereo_frame(const int xeye, const int xoff)
 				ogl_stereo_viewport[0] += ogl_stereo_viewport[2];		// x = w/2
 				break;
 			// half-height viewports for above/below format
+			case STEREO_ABOVE_BELOW_SYNC: {
```

Comment:
This case block looks strange.  By allowing it to fall through, the `-=` on line 1348 effectively undoes the `-= dy` from line 1344:

```
v3 -= dy;
v1 -= dy;
fallthrough;
v1 -= v3;
```

->

```
v1 -= dy;
fallthrough;
v1 -= (v3 - dy);
```

->

```
v1 -= dy;
fallthrough;
v1 += -v3 + dy;
```

So it looks like you could get the same result on v1 by transposing the new `-=` lines, and removing the fallthrough.  Is that right?  What is the advantage of the fallthrough technique here?

Reply:
This adjustment was meant to be symetrical with the viewport height adjustment in the left-eye case. It had been reshuffled from its own separate case before realizing any code similarity with the original above/below format, so this could be probably be simplified further by evaluating`dy` for either case.

Reply:
Hmm, maybe not so obvious now when revisiting. The main distinction between the two above/below formats is that viewport height (`v3`) needs to be adjusted for the sync blanking interval, whereas the viewport Y (`v1`) remains clamped to 0 in both format variants as common case.